src/trade_bot_manager.py

```diff
@@ -0,0 +1,84 @@
+import concurrent.futures
+from src.config import Config
+from src.factories.context_factory import ContextFactory
+from src.interfaces import IContext
+from src.factories.account_factory import AccountFactory
+from src.action_writer import ActionWriter
+from src.ema_strategy import EmaStrategy
+from src.factories.symbol_factory import SymbolFactory
+from src.trade_bot import TradeBot
+from src.factories.trade_executor_factory import TradeExecutionFactory
+from src.constants import production
+from src.ws_server import Messenger, TradeBotWebsocketServer
+
+# This class is a singleton. There should only ever be one trade bot manager
+class TradeBotManager():
+
+    context: IContext
+    first_bot_created: bool
+    id_bot_map: dict
+    messenger: Messenger
+    pool : concurrent.futures.ThreadPoolExecutor
```

Comment:
Do we need to use a thread pool to manage bots?

Assuming bots run indefinitely it seems to defeat the point of using thread pool because we won't be reusing threads.

Reply:
Because users can delete trade bots, they are not necessarily running indefinitely.